### PR TITLE
Fix broken styling in Elementor preview mode. [ECP-1065]

### PR DESCRIPTION
**Pull request by EugeneKyale**
Load the assets required by the `Event View` Elementor widget in preview mode.

Ticket : https://theeventscalendar.atlassian.net/browse/ECP-1065

https://www.loom.com/share/9ce90a65a5604d63aa28dcf209a3ed9a

### CODE REVIEW

File: src/Tribe/Views/V2/Assets.php [hunk 1]
```diff
@@ -528,6 +528,20 @@ public function should_enqueue_frontend() {
...
+		/**
+		 * Checks whether the page is being viewed in Elementor preview mode.
+		 * 
+		 * @since TBD
+		 * 
+		 * @return bool $should_enqueue Should the frontend assets be enqueued.
+		 */
+		if ( is_plugin_active( 'elementor/elementor.php' ) && \Elementor\Plugin::$instance->preview->is_preview_mode() ) {
+			return $this->should_enqueue = true;
+		}
```
Comment (bordoni) [inline src/Tribe/Views/V2/Assets.php:543]: We should use the same conditional as the code from the integrations. To determine if elementor is active.
```
		if ( ! defined( 'ELEMENTOR_PATH' ) || empty( ELEMENTOR_PATH ) ) {
			return;
		}
```

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Tribe/Views/V2/Assets.php (hunk 1 above)
Comment (EugeneKyale) [reply src/Tribe/Views/V2/Assets.php]: @bordoni I've updated the conditional

File: src/Tribe/Views/V2/Assets.php [hunk 2]
```diff
@@ -528,6 +528,17 @@ public function should_enqueue_frontend() {
...
 		}
 
+		/**
+		 * Checks whether the page is being viewed in Elementor preview mode.
+		 * 
+		 * @since TBD
+		 * 
+		 * @return bool $should_enqueue Should the frontend assets be enqueued.
+		 */
+		if ( ! ( ! defined( 'ELEMENTOR_PATH' ) || empty( ELEMENTOR_PATH ) ) && \Elementor\Plugin::$instance->preview->is_preview_mode() ) {
```
Comment (Camwyn) [inline src/Tribe/Views/V2/Assets.php:538]: Are we sure that we should _always_ load this in The Elementor preview? This bypasses **all** the checks in Template_Bootstrap

Comment (Camwyn) [reply src/Tribe/Views/V2/Assets.php:538]: If so - the logic could be simplified a lot.
```suggestion
		if (
			defined( 'ELEMENTOR_PATH' )
			&& ! empty( ELEMENTOR_PATH )
			&& \Elementor\Plugin::$instance->preview->is_preview_mode()
        ) {
```

Comment (Camwyn) [reply src/Tribe/Views/V2/Assets.php:538]: ...and the call to `is_preview_mode` can probably be boiled down to (for our purposes)

`if ( isset( $_GET[ 'elementor-preview' ] ) && $_GET[ 'elementor-preview' ] === get_the_ID() )`

removing the requirement for calling a third-party class?

Sorry, to be clearer, try:
```
if (
	defined( 'ELEMENTOR_PATH' )
	&& ! empty( ELEMENTOR_PATH )
	&& isset( $_GET[ 'elementor-preview' ] )
	&& $_GET[ 'elementor-preview' ] === get_the_ID()
) {
```

**Review by Camwyn — CHANGES_REQUESTED** [review]

File: src/Tribe/Views/V2/Assets.php (hunk 2 above)
Comment (EugeneKyale) [reply src/Tribe/Views/V2/Assets.php]: @Camwyn we need to load it on the elementor preview when the event view widget is added to the page.

Comment (EugeneKyale) [reply src/Tribe/Views/V2/Assets.php]: I've updated the conditional but haven't added the last one i.e. `$_GET[ 'elementor-preview' ] === get_the_ID()`. `isset( $_GET[ 'elementor-preview' ] )` works as expected.

**Review by Camwyn — APPROVED** [review]
🥇

File: src/Tribe/Views/V2/Assets.php (hunk 2 above)
Comment (Camwyn) [reply src/Tribe/Views/V2/Assets.php]: Yeah, the `$_GET[ 'elementor-preview' ] === get_the_ID()` part was pulled directly from `is_preview_mode()` this is really nice and simple now 😄

**Pull request merged**